File: js/async-await/1.js
require("./common")();

let events = [];

function log(event) {
  events.push(event);
}

// Yield from coroutine

function* cc() {
  log("c started");
  yield;
  log("c finished");
}

log("main started");
let c = cc();
c.next();
c.next();
c.next();
log("main finished");

expectToEqual(events, [
  'main started',
  'c started',
  'c finished',
  'main finished'
]);

File: python/async/1.py
from common import expect_to_be_equal

events = []


def log(event):
    events.append(event)


def create_generator():
    log("generator started")
    yield 1
    log("generator finished")


log("main started")

generator = create_generator()
try:
    next(generator)
    next(generator)
except StopIteration:
    log("main StopIteration")

log("main finished")

expect_to_be_equal(events, [
    'main started',
    'generator started',
    'generator finished',
    'main StopIteration',
    'main finished'
])
